### PR TITLE
HV: convert hexadecimals to unsigned

### PR DESCRIPTION
**Pull request by junjiemao1**
Per MISRA C, operands to bit-wise operations should have unsigned
types. However, C99 prioritizes to use signed integers for hexadecimal constants
without the 'u' suffixes, leading to tons of bit operations on signed integers.

This patch series add the 'u' suffixes to the constants which are used in bit
operations, and add the intended width of these integers when applicable
(i.e. the target value is at least 32-bit wide) to avoid functional differences
due to signed vs. unsigned extensions.

### CODE REVIEW

**Comment by NanlinXie** [comment]
Can one of the admins verify this patch?

File: hypervisor/include/arch/x86/cpu.h [hunk 1]
```diff
@@ -41,80 +41,80 @@
 /* Define page size */
 #define CPU_PAGE_SHIFT          12
 #define CPU_PAGE_SIZE           0x1000
-#define CPU_PAGE_MASK           0xFFFFFFFFFFFFF000
+#define CPU_PAGE_MASK           0xFFFFFFFFFFFFF000ULL
```
Comment (dongyaozu) [inline hypervisor/include/arch/x86/cpu.h]: Why this is ULL ? It is uint64_t, right?

File: hypervisor/include/arch/x86/cpu.h [hunk 2]
```diff
@@ -310,7 +310,7 @@ void start_cpus();
 /* This macro writes the stack pointer. */
 #define CPU_SP_WRITE(stack_ptr)                             \
 {                                                           \
-	uint64_t rsp = (uint64_t)stack_ptr & ~(CPU_STACK_ALIGN - 1);  \
+	uint64_t rsp = (uint64_t)stack_ptr & ~(CPU_STACK_ALIGN - 1U);  \
```
Comment (dongyaozu) [inline hypervisor/include/arch/x86/cpu.h]: This should be 1UL.. Not?

File: hypervisor/arch/x86/gdt.c [hunk 3]
```diff
@@ -11,14 +11,14 @@ static void set_tss_desc(union tss_64_descriptor *desc,
 {
 	uint32_t u1, u2, u3;
 
-	u1 = ((uint64_t)tss << 16) & 0xFFFFFFFF;
-	u2 = (uint64_t)tss & 0xFF000000;
-	u3 = ((uint64_t)tss & 0x00FF0000) >> 16;
+	u1 = ((uint64_t)tss << 16) & 0xFFFFFFFFULL;
```
Comment (dongyaozu) [inline hypervisor/arch/x86/gdt.c]: I am lost on the usage of UL/ULL. I will stop here to get it clarified

Comment (junjiemao1) [reply hypervisor/arch/x86/gdt.c]: Per C99, L (long) is at least 32-bit while LL (long long) is at least 64-bit. The actual bits of each integer type is implementation-defined. Thus I use ULL for 64-bit constants whenever possible.

File: hypervisor/include/arch/x86/cpu.h (hunk 2 above)
Comment (junjiemao1) [reply hypervisor/include/arch/x86/cpu.h]: My bad. CPU_STACK_ALIGN and the '1' here should all be uint64_t.

**Comment by lijinxia** [comment]
Verified+1

**Pull request merged**